static/app/components/charts/onDemandMetricRequest.spec.tsx

```diff
@@ -0,0 +1,204 @@
+    partial: false,
+    includeTransformedData: true,
+    sampleRate: SAMPLE_RATE,
+  };
+
+  describe('with props changes', function () {
+    beforeAll(function () {
+      (doEventsRequest as jest.Mock).mockImplementation(() =>
+        Promise.resolve({
+          isMetricsData: true,
+          data: [],
+        })
+      );
+    });
+
+    it('makes requests', async function () {
+      render(<OnDemandMetricRequest {...DEFAULTS}>{mock}</OnDemandMetricRequest>);
+      expect(mock).toHaveBeenNthCalledWith(
+        1,
+        expect.objectContaining({
+          loading: true,
+        })
+      );
+
+      await waitFor(() =>
+        expect(mock).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            loading: false,
+            seriesAdditionalInfo: {
+              current: {
+                isExtrapolatedData: true,
+                isMetricsData: false,
+              },
+            },
+            timeseriesData: [
+              {
+                seriesName: expect.anything(),
+                data: [],
+              },
+            ],
+            originalTimeseriesData: [],
+          })
+        )
+      );
+
+      expect(doEventsRequest).toHaveBeenCalledTimes(2);
+    });
+
+    it('makes a new request if projects prop changes', function () {
+      const {rerender} = render(
+        <OnDemandMetricRequest {...DEFAULTS}>{mock}</OnDemandMetricRequest>
+      );
+      doEventsRequest as jest.Mock;
+
+      rerender(
+        <OnDemandMetricRequest {...DEFAULTS} project={[123]}>
+          {mock}
+        </OnDemandMetricRequest>
+      );
+
+      expect(doEventsRequest).toHaveBeenCalledTimes(2);
+      expect(doEventsRequest).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.objectContaining({
+          project: [123],
+          useOnDemandMetrics: true,
+        })
+      );
+    });
+
+    it('makes a new request if environments prop changes', function () {
+      const {rerender} = render(
+        <OnDemandMetricRequest {...DEFAULTS}>{mock}</OnDemandMetricRequest>
+      );
+      doEventsRequest as jest.Mock;
+
+      rerender(
+        <OnDemandMetricRequest {...DEFAULTS} environment={['dev']}>
+          {mock}
+        </OnDemandMetricRequest>
+      );
+
+      expect(doEventsRequest).toHaveBeenCalledTimes(2);
+      expect(doEventsRequest).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.objectContaining({
+          environment: ['dev'],
+          useOnDemandMetrics: true,
+        })
+      );
+    });
+
+    it('makes a new request if period prop changes', function () {
+      const {rerender} = render(
+        <OnDemandMetricRequest {...DEFAULTS}>{mock}</OnDemandMetricRequest>
+      );
+      (doEventsRequest as jest.Mock).mockClear();
+
+      rerender(
+        <OnDemandMetricRequest {...DEFAULTS} period="7d">
+          {mock}
+        </OnDemandMetricRequest>
+      );
+
+      expect(doEventsRequest).toHaveBeenCalledWith(
```

Comment:
done we need to check for the number of calls here too? `expect(doEventsRequest).toHaveBeenCalledTimes(2);`

Reply:
Yep, we can, added in [d36fc34](https://github.com/getsentry/sentry/pull/53428/commits/d36fc345efd113a59424d724de3c793502c9ad4d)